Bottom: fe4cb17f4c0c26ecadfff46a5a44815176c4acbd
Top:    554e6b222e782e46b98a3cd6e93fd0556bdee98b
Author: Ian Kent <raven@themaw.net>
Date:   2013-03-03 09:43:22 +0800

Refresh of autofs-5.0.7-fix-map-read-fail-incorrectly-set-on-master-re-read.patch

---

diff --git a/modules/lookup_file.c b/modules/lookup_file.c
index b6432ff..411f280 100644
--- a/modules/lookup_file.c
+++ b/modules/lookup_file.c
@@ -397,8 +397,9 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 	unsigned int path_len, ent_len;
 	int entry, cur_state;
 
+	/* Don't return fail on self include, skip source */
 	if (master->recurse)
-		return NSS_STATUS_UNAVAIL;
+		return NSS_STATUS_SUCCESS;
 
 	if (master->depth > MAX_INCLUDE_DEPTH) {
 		error(logopt, MODPREFIX
@@ -452,11 +453,6 @@ int lookup_read_master(struct master *master, time_t age, void *context)
 				     "failed to read included master map %s",
 				     master->name);
 			}
-			/*
-			 * Plus map include failures don't cause the map
-			 * read to fail.
-			 */
-			master->read_fail = 0;
 			master->depth--;
 			master->recurse = 0;
